### ISSUE TITLE
sub-Chapters/Volumes with periods for filenames do not scroll through sequentially when scrolling chapters 

### ISSUE THREAD

**Issue by teekay30** [Kareadita/Kavita#2157]
**If this is a feature request, request [here](https://feats.kavitareader.com/) instead. Feature requests will be deleted from Github.**

Please put as much information as possible to help me understand your issue. OS, browser, version are very important!

**Describe the bug**
When a manga series has (mixed) periods in their volumes the next chapter button does not go in the expected order, for example - Chapter 1, 2, 3.1, 3.2, 3.3, 4.1, 4.2, 5 will potentially go as 1 -> 2 -> 3.3 -> 4.2 -> 5.

If there is another naming convention I should use, please let me know!

**To Reproduce**
1. Have manga chapters with the following volume names, using a folder structure I have for one of my series:

```
vol_001.zip
vol_2.1.zip
vol_2.2.zip
vol_2.3.zip
vol_2.4.zip
vol_2.5.zip
vol_3.1.zip
vol_3.2.zip
vol_3.3.zip
vol_3.4.zip
vol_3.5.zip
vol_3.6.zip
vol_4.1.zip
vol_4.2.zip
vol_4.3.zip
vol_4.4.zip
vol_4.5.zip
vol_4.6.zip
vol_5.1.zip

etc
```

3. Start reading the manga from Chapter 1. This load vol_001 as expected
4. Scroll down to the end of the chapter and let the web viewer navigate to the next chapter or click the 'Next Chapter/Volume' icon or hit the right arrow key on the keyboard to transition to the next chapter/volume.
5. See that the next chapter loaded is 2.5 rather than 2.1

**Expected behavior**
The following chapter loaded should consider the 'dot' or subchapter(?) numbering as well.
I have also tried to renamed the volumes to `vol_002.1.zip`, however it did not seem to respect the ordering.

**Screenshots**
File system: (chapter 001 is way up the screen due to volumes 10+ and 20+)
![image](https://github.com/Kareadita/Kavita/assets/14088402/bd808ff2-a9a4-416f-b07c-68dc827d03de)

**Desktop (please complete the following information):**
 - OS: Docker v24.0.2
 - Browser: Firefox 115.0.2 (64-bit)
 - Version :  0.7.4.0

**Smartphone (please complete the following information):**

**Additional context**
I am using a script to download manga, so if I need to change the way I name stuff so that Kavita can do this without much change I'd like to know what that should be for these small/sub chapters.

**Comment by teekay30** [comment]
Sorry I actually tried a couple more iterations and using dashes seems to do the trick and Kavita successfully merges all 'minor' chapters into one chapter. Example:
```
vol_001.zip
vol_2-1.zip
vol_2-2.zip
vol_003.zip
```

This will create a simple 'Volume 2' with everything in the right order and go from 1 -> 2 -> 3 successfully.

**Comment by majora2007** [comment]
Okay I tested this and it does actually work as it should. This must have been a parsing issue from filename.

**Comment by teekay30** [comment]
sorry what do you mean by it being a 'parsing issue'? In older versions and it's fixed in the current 0.7.5 branch?
I've already changed my downloader to use the dashes as it works, however was curious to know why 'decimal' values didn't seem to work
